specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,302 @@
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring sessionId = GetJSONStringField(jsonMessage.get(), L"sessionId");
+                auto session = m_devToolsSessionMap.find(sessionId);
+                if (session != m_devToolsSessionMap.end())
+                {
+                    m_devToolsTargetDescriptionMap.erase(session->second);
+                    m_devToolsSessionMap.erase(session);
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetDetachedToken));
+    receiver.reset();
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Target.targetInfoChanged", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring targetId = GetJSONStringField(jsonMessage.get(), L"targetId");
+                if (m_devToolsTargetDescriptionMap.find(targetId) !=
+                    m_devToolsTargetDescriptionMap.end())
+                {
+                    // This is a target that we are interested in, update description.
+                    std::wstring type = GetJSONStringField(jsonMessage.get(), L"type");
+                    std::wstring url = GetJSONStringField(jsonMessage.get(), L"url");
+                    m_devToolsTargetDescriptionMap[targetId] = type + L"," + url;
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetInfoChangedToken));
+    m_webView->CallDevToolsProtocolMethod(
+        L"Target.setAutoAttach",
+        LR"({"autoAttach":true,"waitForDebuggerOnStart":false,"flatten":true})", nullptr);
+}
+```
+## WinRT and .NET    
+```c#
+// The sessions and targets descriptions are tracked by 2 dictionaries:
+// SessionId to TargetId dictionary: m_devToolsSessionMap;
+// TargetId to description dictionary, where description is "<target type>,<target url>": m_devToolsTargetInfoMap
+// GetJSONStringField is a helper function that can retrieve a string field from a json message.
+
+private void CoreWebView2_ConsoleAPICalled(CoreWebView2 sender, CoreWebView2DevToolsProtocolEventReceivedEventArgs args)
```

Comment:
It would be nice to include some more comments in the sample code. Especially for the CDP events and methods where folks won't necessarily know what they do.

Reply:
Added some comments.